code/datums/diseases/advance/symptoms/vision.dm

```diff
@@ -88,12 +88,14 @@ Bonus
 	base_message_chance = 50
 	symptom_delay_min = 15
 	symptom_delay_max = 50
-	var/better_vis = FALSE
-	var/nvgs = FALSE
-	var/thermal = FALSE
-	var/xray = FALSE
-	var/nvision_buff = FALSE //Toggles TRUE when the Night Vision Buffs are Applied
-	var/other_buff = FALSE //Toggles TRUE when Thermal/Xray is applied
+	var/list/thresholds = list()//An Associative List of Threshhold related vars. ["better_vis"], ["nvgs"], ["thermal"], ["xray"], ["nvision_buff"], ["other_buff"] are the indexes.
+	//Associative Index Initialization
+	thresholds["better_vis"] = FALSE
+	thresholds["nvgs"] = FALSE
+	thresholds["thermal"] = FALSE
+	thresholds["xray"] = FALSE
+	thresholds["nvision_buff"] = FALSE //Toggles TRUE when the Night Vision Buffs are Applied
+	thresholds["other_buff"] = FALSE //Toggles TRUE when Thermal/Xray is applied
```

Comment:
There is no need for this. If you check this and it wasn't set before it will return null which is essentially false.